### PR TITLE
automated plugins loader

### PR DESCRIPTION
**Pull request by AlhasanIQ**

### CODE REVIEW

**Comment by AlKass** [comment]
I think I can make the case against automating this part.
As a matter of fact, I initially had it automated then a few thoughts came to mind:

* We're designing this tool with a lot of plugins that almost no user will have the need to utilize them all. If we iterate through all existing packages and 'import' them, the user pretty much has no control over disabling any of the existing plugins.

* Using the name of the directory as the name of the plugin might be upsetting to some users. One instance I can think of is when you'd have a descriptive plugin name but want to use an abbreviation in the API URL.

* If the tool sees some popularity over time and people start writing plugins and sharing them as GitHub repos, a possible naming convention might be something like pitrol-plugin-name. `API/pitrol-pulgin-name/feature' won't look very professional.

Long story short. I would highly prefer utilizing a loading method that cares about user's preferences. If you could modify this loading mechanism to achieve this goal, that would be great.

Fadi

**Comment by AlhasanIQ** [comment]
it does the job for now,

when we have the Database/Datastore set up we will do some checking before the `require` statement

**Comment by AlKass** [comment]
cool then. merged

**Comment by AlKass** [comment]
I just pushed two bug fixes related to this merge. For future reference, bare the following in mind:

* A single plugin can contain any number of features, so to reference a specific feature over the API, we use the following format:

`
/API/<Plugin Name>/<Feature Name>
`

The merged code was using the first feature's name as the name of the plugin which caused the plugin handler to misbehave.
